src/modules/auth/auth-passport-jwt/controller/jwtAuthController.ts

```diff
@@ -0,0 +1,114 @@
+import { Request, Response } from 'express';
+import bcrypt from 'bcrypt';
+import UserModel from '../../../users/model/UserModel';
+import { blacklistToken, generateJWT } from '../../../../middleware/passport/strategies/jwt/jwt-utils';
+import jwt from 'jsonwebtoken';
+import UserDao from "../../../users/dao/UserDao";
+
+/**
+ * Register a new user.
+ *
+ * This function handles user registration. It hashes the user's password,
+ * saves the user to the database, generates a JWT token for the user,
+ * blacklists the token, and returns the token in the response.
+ *
+ * @param {Request} req - The Express request object.
+ * @param {Response} res - The Express response object.
+ */
+export const register = async (req: Request, res: Response) => {
+    const { username, email, passwordHash: hashPassword, role, firstName, lastName, address, phone } = req.body;
+    try {
+        // Hash the provided password
+        const passwordHash = await bcrypt.hash(hashPassword, 10);
+
+        // Initialize the UserDao with the UserModel
+        const userDao = new UserDao(UserModel);
+
+        // Create the user in the database
+        const userData = await userDao.createUser({
+            username,
+            email,
+            passwordHash,
+            role,
+            firstName,
+            lastName,
+            address,
+            phone
+        });
+
+        // Generate a JWT token for the newly created user
+        const token = generateJWT(userData);
+        const expiryDate = new Date(Date.now() + 3600 * 1000); // 1 hour
+
+        // Blacklist the token (optional step depending on your use case)
+        await blacklistToken(token, expiryDate);
+
+        if (userData) {
+            res.status(201).json({ token });
+        }
+    } catch (err) {
+        res.status(500).json({ error: 'Error registering user' });
+    }
+};
+
+/**
+ * Log in an existing user.
+ *
+ * This function handles user login. It verifies the user's email and password,
+ * generates a JWT token if the credentials are valid, and returns the token in
+ * the response.
+ *
+ * @param {Request} req - The Express request object.
+ * @param {Response} res - The Express response object.
+ */
+export const login = async (req: Request, res: Response) => {
+    try {
+        const { email, password } = req.body;
+
+        // Initialize the UserDao with the UserModel
+        const userDao = new UserDao(UserModel);
+
+        // Find the user by email
+        const user = await userDao.findUserByEmail(email);
+
+        // Check if the user exists and the password is valid
+        if (!user || !(await bcrypt.compare(password, user.passwordHash))) {
+            return res.status(400).json({ error: 'Invalid email or password' });
+        }
+
+        // Generate a JWT token for the user
+        const token = generateJWT(user);
```

Comment:
Don't we need to same generated token in the blacklist ?

Reply:
no, blacklisted should be only the tokens which are logout, for login and registrations  just generating tokens and sent as a response

Reply:
maybe I don't understand how that works. how to we check if the token is valid if we are not saving it anywhere?